sdk/tables/azure-data-tables/azure/data/tables/_models.py

```diff
@@ -339,6 +341,7 @@ def __init__(self, command, table, **kwargs):
         self.filter = kwargs.get("filter")
         self.select = kwargs.get("select")
         self.location_mode = None
+        self.entity_hook = kwargs.pop("entity_hook", _convert_to_entity)
```

Comment:
I don't think this will work, as currently `entity_hook` is always passed in with the value of `None` from the client. So I think we'll need:
`self.entity_hook = kwargs.pop("entity_hook", None) or _convert_to_entity`